### PR TITLE
sign in api

### PR DESCRIPTION
**Pull request by cpt2162**

### CODE REVIEW

File: api/models/userModel.js [hunk 1]
```diff
@@ -1,16 +1,32 @@
 const db = require('../db');
+const crypto = require('crypto');
```
Comment (heathermoses) [inline api/models/userModel.js:2]: Is this the built in crypto package? I was looking at the documentation for crypto and it says it's deprecated so just wanted to confirm
This is the link to the documentation I was looking at: https://www.npmjs.com/package/crypto

File: api/routes/userRouter.js [hunk 2]
```diff
@@ -4,22 +4,27 @@ const userModel = require('../models/userModel');
...
-            throw new Error("Missing required user parameters: " + missingRequiredParams.join(', '));
+        const createParams = ['username', 'password', 'first_name', 'last_name', 'phone'];
+        const signInParams = ['username', 'password'];
+
+        const missingCreateParams = createParams.filter(param => !(param in req.body));
+        const missingSignInParams = signInParams.filter(param => !(param in req.body));
+
+        if(missingCreateParams.length == 0){
+            user = await userModel.createUser(req.body);
+            res.status(200).json({ user_id: user.id, message: "User created successfully with id " + user.id + "." });
```
Comment (heathermoses) [inline api/routes/userRouter.js:20]: What does `.status(200)` do? Does that force the result to have a status of 200?

File: api/sql/userSQL/initialize.sql [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
   data_id INT NOT NULL,
   FOREIGN KEY (data_id) REFERENCES UserData (id)
+);
+
+CREATE TABLE UserSessions (
+  id SERIAL PRIMARY KEY,
+  session_key VARCHAR (100) NOT NULL,
+  user_id INT NOT NULL,
+  exp_date TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP + interval '4 hours',
+  FOREIGN KEY (user_iD) REFERENCES Users (id)
```
Comment (heathermoses) [inline api/sql/userSQL/initialize.sql:17]: Does `user_iD` need to be `user_id`?

File: api/sql/userSQL/initialize.sql [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
   password TEXT NOT NULL,
   data_id INT NOT NULL,
   FOREIGN KEY (data_id) REFERENCES UserData (id)
+);
+
+CREATE TABLE UserSessions (
+  id SERIAL PRIMARY KEY,
+  session_key VARCHAR (100) NOT NULL,
+  user_id INT NOT NULL,
+  exp_date TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP + interval '4 hours',
```
Comment (heathermoses) [inline api/sql/userSQL/initialize.sql:16]: What does the `+ interval '4 hours'` do?

**Review by heathermoses — CHANGES_REQUESTED** [review]
I think there might be one typo which is why I'm requesting a change but if it's actually not a typo, then everything else looks good to me and I'll approve! I left a couple comments just to better understand the code here.

**Review by heathermoses — APPROVED** [review]

File: api/routes/userRouter.js (hunk 2 above)
Comment (cpt2162) [reply api/routes/userRouter.js:20]: It just sets the status code of the response to 200

File: api/sql/userSQL/initialize.sql (hunk 4 above)
Comment (cpt2162) [reply api/sql/userSQL/initialize.sql:16]: this just adds four hours to the current timestamp. its similar timedelta() in python. I wasn't sure how long we want session tokens to be active, we can change this if need be

File: api/models/userModel.js (hunk 1 above)
Comment (cpt2162) [reply api/models/userModel.js:2]: Yes I didn't add any dependencies, this was already built in

File: api/sql/userSQL/initialize.sql (hunk 3 above)
Comment (cpt2162) [reply api/sql/userSQL/initialize.sql:17]: SQL with interpret it all lowercase. Should be changed for consistency purposes though

**Pull request merged**